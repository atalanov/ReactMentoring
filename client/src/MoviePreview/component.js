import React from 'react';
import MoviePreviewImage from '../MoviePreviewImage';
import MoviePreviewDescription from '../MoviePreviewDescription';

export default props => (
  <div>
    <MoviePreviewImage src={props.item.posterUrl} />
    <MoviePreviewDescription
      jenre={props.item.jenre}
      year={props.item.year}
      name={props.item.name}
    />
  </div>
);

