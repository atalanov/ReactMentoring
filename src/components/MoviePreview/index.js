import React from 'react';
import { MovieImage, MovieDescription } from '../index';

const MoviePreview = props => (
  <div>
    <MovieImage src={props.item.posterUrl} />
    <MovieDescription
      jenre={props.item.jenre}
      year={props.item.year}
      name={props.item.name}
    />
  </div>
);

export default MoviePreview;
