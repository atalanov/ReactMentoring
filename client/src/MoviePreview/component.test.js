import React from 'react';
import { shallow, render } from 'enzyme';
import MoviePreview from './component';

describe('MoviePreview tests', () => {
    test('Component MoviePreview renders correctly', () => {
        const item = {
            name: 'TestFilm',
            jenre: 'Horror',
            year: '2018',
            posterUrl: '/testurl',
        };
        const component = shallow(<MoviePreview item={item} />);
        expect(component).toMatchSnapshot();
    });
});
