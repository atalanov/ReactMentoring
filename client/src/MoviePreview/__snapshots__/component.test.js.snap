// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePreview tests Component MoviePreview renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    item={
      Object {
        "genres": Array [
          "Horror",
        ],
        "poster_path": "/testurl",
        "release_date": "2018",
        "title": "TestFilm",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          id={undefined}
          src="/testurl"
        />,
        <Unknown
          id={undefined}
          jenre="Horror"
          name="TestFilm"
          year="2018"
        />,
      ],
      "className": "col col-md-4",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": undefined,
          "src": "/testurl",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": undefined,
          "jenre": "Horror",
          "name": "TestFilm",
          "year": "2018",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            id={undefined}
            src="/testurl"
          />,
          <Unknown
            id={undefined}
            jenre="Horror"
            name="TestFilm"
            year="2018"
          />,
        ],
        "className": "col col-md-4",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": undefined,
            "src": "/testurl",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": undefined,
            "jenre": "Horror",
            "name": "TestFilm",
            "year": "2018",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
